cli/cli.go

```diff
@@ -53,6 +55,35 @@ func (c *CLI) ReadEnv() {
 	}
 }
 
+const ContextFileName = ".hcloud-context"
```

Comment:
shouldn't the file be named `.hcloud`. in the future there maybe will be more directory specific configuration beside the context.
```suggestion
const ContextFileName = ".hcloud"
```

Reply:
Valid point. However, I would prefer to make using that feature as easy as possible and not require the user to write TOML/YAML/JSON files:

    echo my-context > .hcloud-context

If you need per-directory configs, maybe you should use tools like [direnv](https://direnv.net) which allows you to set per-directory environment variables. By setting `HCLOUD_CONFIG` you can configure `hcloud` to use a different config file per directory:

```
$ cat .envrc
export HCLOUD_CONFIG=.hcloud.toml
```

Actually, you could even use direnv to replace this pull request:

```
$ cat .envrc
export HCLOUD_CONTEXT=my-context
```

To be honest, I tend to prefer the generic direnv solution over our custom implementation.

Reply:
> To be honest, I tend to prefer the generic direnv solution over our custom implementation.

this would be again, another dependency. the stack grows and grows (vagrant, puppet, terraform, hcloud). i vote strongly against using direnv.
your solution simply works. it has minimal performance impact but if a file is there it is read.

symfony (php framework) made the same step, first it removed `parameters.yaml` to move to `.env` but `.env` was only used for dev not for production. then the move "back". refactor of the `.env` component and also the official statement, that it can be used in production.

having this config file hurts nobody and gives another option without adding another tool to the toolchain but it don't prevent you from using direnv.

we are software architects, the big advantage over building architects is, that we can have both a "flat top" and a "pitched roof" at the same time.

Reply:
just had a look at direnv, it doesn't look like to have full windows support. if i don't use bash i can't do a `eval "$(direnv hook bash)"` so it won't run as transparent as a simple config file.
often i cannot easily control which shell a process spans. 

- will terraform span a bash shell where it will load my profile where the direnv is loaded?
- will the direnv smoothly work within a bash shell called with terraform on windows?
- how will hooking behave if called from terrafrom?

sounds like a lot of overhead, config and testing for simple context switching.

having a full `TOML/YAML/JSON` file with a full possibilities for configuring hcloud would be awesome. it makes a really good developer expierence without killing performance. the only performance overhead is one `if file_exists()` check. no pain and much gain :)